AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,109 @@
 //
 
 import UIKit
+import AVKit
+import AVFoundation
+import MobileCoreServices
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    //MARK: - Properties
     var titleForCell: String = "Double Video"
-
+    var imagePickerController: UIImagePickerController!
+    var videoURL: URL?
+    var playerA: AVPlayer?
+    var playerB: AVPlayer?
+    
+    //MARK: - Outlets
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
+    //MARK: - Methods
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
-    override func viewDidLayoutSubviews() {
-        // update your layers' frames here
+    
+    //    override func viewDidLayoutSubviews() {
+    //        // update your layers' frames here
+    //        //this isn't actually doing anything
+    //    }
+    
+    func handleVideoPlaying() {
+        if let url = self.videoURL {
+            let asset = AVAsset(url: url)
+            let playerItem = AVPlayerItem(asset: asset)
+            let player = AVPlayer(playerItem: playerItem)
+            let playerLayer = AVPlayerLayer(player: player)
+            
+            //self.player.rate == 0.0 did not lead to a positive response so I am successfully using self.player == nil
+            if self.playerA == nil {
+                //bonus
+                self.videoContainerTop.layer.sublayers?.forEach { (layer) in
```

Comment:
I don't think this ever got to run because the block isn't entered after playerA is set. Also this comparison:

```swift
if layer == AVPlayerLater() {
```
will always fail because ```layer``` is one instance and AVPlayerLayer() will always construct a new different one. I think you meant:

```swift
if layer is AVPlayerLayer
```